docs/install/airgapped-clusters.md

```diff
@@ -0,0 +1,365 @@
+
+REGISTRY_IMG_PROXY=my-registry-url/quay.io/brancz/kube-rbac-proxy:v0.10.0
+sed -i -e "s#image: quay.io/brancz/kube-rbac-proxy:v0.10.0#image: $REGISTRY_IMG_PROXY#g" ondat-operator.yaml
+
+# Check the change
+grep -C2 "image:" ondat-operator.yaml
+```
+
+## Step 3. Defining StorageOSCluster CR
+
+* The StorageOSCluster definition depends on your cluster. For all available options, check the [operator
+reference](docs/reference/cluster-operator/configuration). For airgapped clusters, it is important to note that `spec.images`
+section needs to be populated with the images from the configMap from Step 1. 
+
+* The file for the `ondat-cluster.yaml` should have the Secret called
+`storageos-api` and the StorageOSCluster definition.
+
+Edit the variable `REGISTRY`
+
+```yaml
+# Set the registry URL
+REGISTRY=my-registry-url
+
+cat <<END > ondat-cluster.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: storageos-api
+  namespace: storageos
+  labels:
+    app: storageos
+type: Opaque
+data:
+  password: c3RvcmFnZW9z # echo -n <username> | base64
+  username: c3RvcmFnZW9z # echo -n <username> | base64
+---
+# CR cluster definition
+apiVersion: storageos.com/v1
+kind: StorageOSCluster
+metadata:
+  name: storageos-cluster
+  namespace: "storageos"
+spec:
+  secretRefName: "storageos-api"
+  secretRefNamespace: "storageos"
+  k8sDistro: "upstream"
+  storageClassName: storageos
+  images:
+    nodeContainer: $REGISTRY/storageos/node:v2.5.0
+    apiManagerContainer: $REGISTRY/storageos/api-manager:v1.2.2
+    initContainer: $REGISTRY/storageos/init:v2.1.0
+    csiNodeDriverRegistrarContainer: $REGISTRY/quay.io/k8scsi/csi-node-driver-registrar:v2.1.0
+    csiExternalProvisionerContainer: $REGISTRY/storageos/csi-provisioner:v2.1.1-patched
+    csiExternalAttacherContainer: $REGISTRY/quay.io/k8scsi/csi-attacher:v3.1.0
+    csiExternalResizerContainer: $REGISTRY/quay.io/k8scsi/csi-resizer:v1.1.0
+    csiLivenessProbeContainer: $REGISTRY/quay.io/k8scsi/livenessprobe:v2.2.0
+    kubeSchedulerContainer: $REGISTRY/k8s.gcr.io/kube-scheduler:v1.21.5
+  kvBackend:
+    address: "storageos-etcd-client.storageos:2379"
+#  nodeSelectorTerms:
+#    - matchExpressions:
+#      - key: "node-role.kubernetes.io/worker" # Compute node label will vary according to your installation
+#        operator: In
+#        values:
+#        - "true"
+END
+```
+
+> ⚠️  The `kubeSchedulerContainer` version depends on the Kubernetes version your
+> cluster is running. In this example v1.21.5. Adjust the tag accordingly. 
+
+If you are using an external Etcd, skip Step 4.
+
+## Step 4. (If Etcd in Kubernetes) Retrieving and amending the YAML for Etcd operator and Etcd CR
+
+1. The YAMLs for Etcd are generated for every release of the product. It is
+best to retrieve the YAML from the machine generated pipeline. To do so, run the following container that prints them on stdout.
+
+```
+ETCD_OPERATOR_VERSION=develop
```

Comment:
Everything else is versioned, is `develop` reasonable or should users use a specific pinned version?

Reply:
I'd prefer a version defined, but I'm afraid that there is no release for it yet. Probably the reason is that it is still considered beta. I'm asking to get a version tagged if possible. I assume that once it is release as GA, it would be tagged.